monai/networks/nets/quicknat.py

```diff
@@ -0,0 +1,447 @@
+        un_pool = nn.MaxUnpool2d(kernel_size=pool, stride=stride_pool)
+
+        # sequence of convolutional strides (like in UNet) not needed as they are always stride_conv. This defaults to 1.
+        def _create_model(layer: int) -> nn.Module:
+            """
+            Builds the QuickNAT structure from the bottom up by recursing down to the bottelneck layer, then creating sequential
+            blocks containing the decoder, a skip connection around the previous block, and the encoder.
+            At the last layer a classifier block is added to the Sequential.
+
+            Args:
+                layer = inversproportional to the layers left to create
+            """
+            subblock: nn.Module
+            if layer < 4:
+                subblock = _create_model(layer + 1)
+
+            else:
+                subblock = Bottleneck(num_filters, se_layer, dropout_layer, max_pool, un_pool, kernel_size, num_filters)
+
+            if layer == 1:
+                down = ConvConcatDenseBlock(num_channels, se_layer, dropout_layer, kernel_size, num_filters)
+                up = ConvConcatDenseBlock(num_filters * 2, se_layer, dropout_layer, kernel_size, num_filters)
+                classifier = ClassifierBlock(2, num_filters, num_classes, stride_conv, kernel_c)
+                return SequentialWithIdx(down, SkipConnectionWithIdx(subblock), up, classifier)
+            else:
+                up = Decoder(num_filters * 2, un_pool, se_layer, dropout_layer, kernel_size, num_filters)
+                down = Encoder(num_filters, max_pool, se_layer, dropout_layer, kernel_size, num_filters)
+                return SequentialWithIdx(down, SkipConnectionWithIdx(subblock), up)
+
+        self.model = _create_model(1)
+
+    def get_selayer(self, n_filters, se_block_type="None"):
+        """
+        Returns the SEBlock defined in the initialization of the QuickNAT model.
+
+        Args:
+            n_filters: encoding half of the layer
+            se_block_type: defaults to None. Valid options are None, CSE, SSE, CSSE
+        Returns: Appropriate SEBlock. SSE and CSSE not implemented in Monai yet.
+        """
+        if se_block_type == "CSE":
+            return se.ChannelSELayer(2, n_filters)
+        # not implemented in squeeze_and_excitation in monai use other squeeze_and_excitation import:
+        elif se_block_type == "SSE" or se_block_type == "CSSE":
+            # Throw error if squeeze_and_excitation is not installed
+            if not flag:
+                raise ImportError("Please install squeeze_and_excitation locally to use SpatialSELayer")
+            if se_block_type == "SSE":
+                return se1.SpatialSELayer(n_filters)
+            else:
+                return se1.ChannelSpatialSELayer(n_filters)
+        else:
+            return None
+
+    # TODO: Do I include this:
+    def enable_test_dropout(self):
```

Comment:
What is the purpose of this method? If it's needed it looks like something that could be done as a function in the test scripts instead, this would help clean up code here a bit.

Reply:
the enable_test_dropout() method iterates over Quicknet's encode and decode blocks, and sets the drop_out attribute of each block to train mode. By doing so, dropout is enabled for these blocks during testing or inference, allowing uncertainty estimation. I think in the original Quicknat implementation it was used to prevent overfitting. I did not use it in my training and was not sure whether to include it for consistency - but it is not needed.

Reply:
If it's not used perhaps it should be removed and a comment made in the docstring for this class that the original implementation used this mechanism. If dropout layers are the only stochastic components of this network then calling the `train()` method instead of `eval()` in testing or inference would have the same effect.

Reply:
You are right! thank you.